backend/functions/lib/adapters/services/logger.service.ts

```diff
@@ -53,4 +53,26 @@ export default class log {
   ) {
     log.logMessage(context, 'debug', namespace, message, data);
   }
+
+  private readonly functionContext: ApplicationContext;
+
+  constructor(functionContext: ApplicationContext) {
+    this.functionContext = functionContext;
+  }
+
+  public info(namespace: string, message: string, data?: unknown) {
+    log.info(this.functionContext, namespace, message, data);
+  }
+
+  public warn(namespace: string, message: string, data?: unknown) {
+    log.warn(this.functionContext, namespace, message, data);
+  }
+
+  public error(namespace: string, message: string, data?: unknown) {
+    log.error(this.functionContext, namespace, message, data);
+  }
+
+  public debug(namespace: string, message: string, data?: unknown) {
+    log.debug(this.functionContext, namespace, message, data);
+  }
```

Comment:
@jamesobrooks I have yes. This will be later in the implementation of logging and we will need to consider this for all functions we wish to capture requests for. Something along these lines: https://medium.com/@WorkloadRancher/step-by-step-add-application-insights-to-azure-function-http-triggers-8d10d19c21fc#:~:text=7.%20Click%20create-,Step,-5%20%E2%80%94%20Instrument%20your

Obviously this is C# code in this example but a similar implementation would be used for our node functions

Reply:
In short, the purpose of this changes was to build our logging object at the start and pass (or inject) the dependency to any other class that would need it. It does remove the need to always pass the application context around.